massa-execution-exports/src/execution_info.rs

```diff
@@ -0,0 +1,343 @@
+#![allow(dead_code)]
+
+use std::collections::HashMap;
+
+use massa_async_pool::AsyncMessageId;
+use massa_hash::Hash;
+use massa_models::block_id::BlockId;
+use massa_models::config::{GENESIS_TIMESTAMP, T0, THREAD_COUNT};
+use massa_models::denunciation::DenunciationIndex;
+use massa_time::MassaTime;
+use schnellru::{ByLength, LruMap};
+
+use massa_deferred_calls::DeferredCall;
+use massa_models::address::Address;
+use massa_models::amount::Amount;
+use massa_models::slot::Slot;
+use serde::Serialize;
+
+use crate::types_trace_info::ExecutionResult;
+// #[cfg(feature = "execution-trace")]
+// use crate::Transfer;
+
+/// Struct for Execution info per slot
+pub struct ExecutionInfo {
+    /// Map of execution info
+    pub info_per_slot: LruMap<Slot, ExecutionInfoForSlot>,
+}
+
+impl ExecutionInfo {
+    /// Create a new ExecutionInfo
+    pub fn new(max_slot_size_cache: u32) -> Self {
+        Self {
+            info_per_slot: LruMap::new(ByLength::new(max_slot_size_cache)),
+        }
+    }
+
+    /// Save transfer for a given slot
+    pub fn save_for_slot(&mut self, slot: Slot, info: ExecutionInfoForSlot) {
+        self.info_per_slot.insert(slot, info);
+    }
+}
+
+/// Struct to store Roll related operation
+#[derive(Debug, Clone)]
+pub enum OperationInfo {
+    /// Roll buy amount
+    RollBuy(Address, u64),
+    /// Roll sell amount
+    RollSell(Address, u64),
+}
+
+/// representation of a transfer
+#[derive(Clone, Debug, Serialize)]
+pub struct TransferHistory {
+    /// sender address
+    pub from: Option<Address>,
+    /// destination address
+    pub to: Option<Address>,
+    /// amount of mas
+    pub amount: Option<Amount>,
+    /// amount of rolls
+    pub roll_count: Option<u64>,
+    /// Transfer context
+    pub context: TransferContext,
+    /// Transfer type (mas, roll, deferred-credits)
+    pub t_type: TransferType,
+    /// Transfer id
+    pub id: Option<String>,
+    /// denunciation index
+    pub denunciation_index: Option<DenunciationIndex>,
```

Comment:
why not include this in the context ? eg. only when there is a slash

Reply:
We can

Reply:
we should :)

Reply:
done